config/src/main/java/com/alibaba/nacos/config/server/service/repository/extrnal/ExternalStoragePersistServiceImpl.java

```diff
@@ -2786,5 +2786,22 @@ public int tenantInfoCountByTenantId(String tenantId) {
         }
         return result.intValue();
     }
-    
+
+    @Override
+    public List<ConfigInfoWrapper> queryConfigInfoByNamespace(String tenant) {
+        Assert.hasText(tenant, "tenant can not be null");
+        String tenantTmp = StringUtils.isBlank(tenant) ? StringUtils.EMPTY : tenant;
+        try {
+            return this.jt.query(
+                    "SELECT data_id,group_id,tenant_id,app_name,type FROM config_info_beta WHERE tenant_id=?",
+                    new Object[]{tenantTmp},
+                    CONFIG_INFO_WRAPPER_ROW_MAPPER);
+        } catch (EmptyResultDataAccessException e) { // Indicates that the data does not exist, returns null.
+            return null;
```

Comment:
Return empty list will be better.

Reply:
why?

Reply:
frontend not do return empty list and search keyword

Reply:
i know。

Reply:
![](https://raw.githubusercontent.com/share-framework/sharer-pic/master/20210831192018.png)

select add showSearch property